cernopendata_client/downloader.py

```diff
@@ -179,74 +270,96 @@ def downloader_file_checker(file_location, file_dest):
     return False
 
 
-def download_single_file(path=None, file_location=None, protocol=None):
-    """Download single file."""
+def download_single_file(
+    path=None, file_location=None, protocol=None, download_engine=None
+):
+    """Download a single file.
+
+    :param path: Directory where file is downloaded
+    :param file_location: Remote location of a file
+    :param protocol: Protocol to be used for downloading a file
+    :param download_engine: Library to be used in downloading files
+    :type path: str
+    :type file_location: str
+    :type protocol: str
+    :type download_engine: str
+
+    :return: None
+    :rtype: None
+    """
     file_name = file_location.split("/")[-1]
     file_dest = path + "/" + file_name
+    download_engine_map = {
+        "requests": requests_available,
+        "pycurl": pycurl_available,
+        "xrootdpyfs": xrootd_available,
+    }
+    download_engine_protocol_http_map = ["pycurl", "requests"]
+    download_engine_protocol_xrootd_map = ["xrootdpyfs"]
+    if download_engine:
+        if not download_engine_map.get(download_engine):
+            display_message(
+                msg_type="error",
+                msg="{} is not installed on system. Please install it.".format(
+                    download_engine
+                ),
+            )
+            sys.exit(1)
     if protocol in ["http", "https"]:
-        if pycurl_available:
-            file_status = downloader_file_checker(file_location, file_dest)
-            if file_status:
-                file_size_offline = os.path.getsize(file_dest)
-                display_message(
-                    msg_type="note",
-                    msg="File {} is incomplete. Resuming download.".format(
-                        file_name,
-                    ),
-                )
-                downloader_pycurl(
-                    file_location, path, file_size_offline=file_size_offline, mode="ab"
-                )
-            else:
-                downloader_pycurl(file_location, path, mode="wb")
+        if download_engine not in download_engine_protocol_http_map:
+            display_message(
+                msg_type="error",
+                msg="{} is not compatible with {} protocol. Please use requests/pycurl engine.".format(
```

Comment:
"Please use requests/pycurl" could be misunderstood.  Better say "Please use 'requests' or 'pycurl'."

Reply:
BTW, in order to have this extensible, perhaps you can have a `config.py` variable which protocol supports which download engine options, and print the message dynamically from that config variable?

(E.g. this will come handy when we shall add vanilla 'xrdcp' as opposed to 'xrootdpyfs'.)